tests/test_crash_similarity.py

```diff
@@ -60,12 +59,42 @@ def test_order_similarity(self):
         doc_mid1, doc_mid2, dist_mid = similarity_mid[0]
         doc_end1, doc_end2, dist_end = similarity_end[0]
 
-        if dist_mid >= dist_end:
-            print('{}\n{}\n{}'.format(doc_mid1, doc_mid2, dist_mid))
-            print('{}\n{}\n{}'.format(doc_end1, doc_end2, dist_end))
-
         self.assertTrue(dist_mid < dist_end)
-        print('\n')
+
+    def test_clean_func(self):
+        for f, expected in funcs:
+            self.assertEqual(crash_similarity.clean_func(f), expected)
+
+    def test_preprocess_returns_clean_stack_traces(self):
+        resp = crash_similarity.preprocess(stack_trace)
+        self.assertEqual(len(resp), 10)
+        self.assertEqual(resp, clean_stack_trace)
+
+    def test_should_skip_returns_correct_boolean_value(self):
+        for f in funcs:
+            if 'xul.dll@' in f or 'XUL@' in f or 'libxul.so@' in f:
+                self.assertTrue(crash_similarity.should_skip(f))
+            else:
+                self.assertFalse(crash_similarity.should_skip(f))
+
+    def test_read_corpus(self):
+        resp = crash_similarity.read_corpus(self.paths)
+        self.assertEqual(type(resp), list)
+        self.assertEqual(len(resp), 378)
+
+    def test_get_stack_traces_for_signature(self):
+        signature = 'js::GCMarker::processMarkStackTop'
+        resp = crash_similarity.get_stack_traces_for_signature(self.paths, signature)
+        for line in utils.read_files(self.paths):
+            data = json.loads(line)
+            if data['signature'] == signature:
+                assert data['proto_signature'] in resp
+
+    def test_get_stack_trace_for_uuid(self):
+        uuid = '90dcebb6-f711-4a8b-9e68-5abf72161109'
```

Comment:
Crash reports are deleted from Socorro after a while (6 months), so we will need to mock the Socorro reply here (https://requests-mock.readthedocs.io/en/latest/overview.html, https://requests-mock.readthedocs.io/en/latest/mocker.html).